mustang_codegen/lib/src/service_generator/service_method_override_visitor.dart

```diff
@@ -85,14 +85,18 @@ class ServiceMethodOverrideVisitor extends SimpleElementVisitor<void> {
       String declaration =
           element.declaration.getDisplayString(withNullability: false);
 
-      overrides.add('''
+      if (beforeHooks.isNotEmpty ||
+          aroundHooks.isNotEmpty ||
+          afterHooks.isNotEmpty) {
```

Comment:
this was a bug

Reply:
it doesn't. But with a new annotation (`@BatchRun`) it starts generating redundant overrides